### PR TITLE
Updates to CashShuffle integration

### PR DESCRIPTION
**Pull request by alwaysAn0n**
This is the first stable update since the original integration.  It includes the necessary changes to accommodate the new wallet backend.  It also includes the requested UI / UX changes discussed on Trello.  Finally, it fixes a few unrelated bugs that I stumbled upon in the process.

### CODE REVIEW

File: www/views/preferencesCashShuffle.html [hunk 1]
```diff
@@ -1,35 +1,34 @@
...
     <div class="list">
 
+      <div class="comment" translate>
+        <img src="../img/obfuscate.svg" alt="CashShuffle" class="cs-img">
+        <div class="settings-description-cs" translate>
+          <p>CashShuffle is a fully decentralized coin mixing protocol that shuffles your Bitcoin Cash with other network participants. This shuffling process obscures your real spending and makes it difficult for chain analysis companies to follow your transactions.</p>
+            <p><strong>Warning: Privacy mode is in beta.</strong> Shuffling your wallet can take hours to complete. Do not enable if you need to spend right away.
+            </p>
+            <div class="cs-button">
+            <a href="https://www.youtube.com/watch?v=8g0Levpj1I4" class="button-green" target="_blank">Learn more</a>
```
Comment (holemcross) [inline www/views/preferencesCashShuffle.html]: Link does not function on iOS. Will need to use external link module.

File: src/js/controllers/create.js [hunk 2]
```diff
@@ -147,7 +147,7 @@ angular.module('copayApp.controllers').controller('createController',
     $scope.create = function() {
 
       // Reject reserved wallet names
-      if($scope.formData.walletName == "CashShuffle Spending Wallet") {
+      if($scope.formData.walletName == 'Private Spending Wallet') {
```
[the remark this answers is not in the document]
Comment (alwaysAn0n) [reply src/js/controllers/create.js]: I would be happy with this change in theory.  Making the service accessible in all the places this string is referenced has other implications though and I don't have time at the moment to do it properly.  I will gladly merge it if someone else is able to do the leg work.

File: www/views/preferencesCashShuffle.html (hunk 1 above)
Comment (alwaysAn0n) [reply www/views/preferencesCashShuffle.html]: @maplesyrupsucker

File: src/js/services/profileService.js [hunk 3]
```diff
@@ -59,15 +59,28 @@ angular.module('copayApp.services')
...
-        if (wallet.coin === 'bch' && wallet.name === 'CashShuffle Spending Wallet') {
-          console.log('Set CashShuffle color to #0073FF'); // Cornflower Blue
-          wallet.color = '#0073FF';
-          wallet.isCashShuffleWallet = true;
+        if (wallet.coin === 'bch') {
+
+          // If this is our CashShuffle spend-only wallet, make it
+          // green and add a flag stating so.
+          if (wallet.name === 'Private Spending Wallet') {
+            console.log('Set CashShuffle color to #0073FF'); // Cornflower Blue
```
[the remark this answers is not in the document]
Comment (alwaysAn0n) [reply src/js/services/profileService.js]: Fixed with f032d7addfd3ae7ea24d465e83da6a0e562c5f63

File: src/js/services/walletService.js [hunk 4]
```diff
@@ -786,43 +785,113 @@ angular.module('copayApp.services').factory('walletService', function($log, $tim
...
 
+      lodash.extend(changeAddress, {
+        legacyAddress: legacyAddress
       });
+
+      return cb(null, changeAddress);
     });
 
+
+    // root.getMainAddresses(someWallet, {}, function(err, allWalletAddresses) {
```
[the remark this answers is not in the document]
Comment (alwaysAn0n) [reply src/js/services/walletService.js]: Resolved with 28d397e56373d822236a648b4600b0cf4e19f0bc

**Pull request merged**